packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -75,50 +77,54 @@ export class HostedPluginSupport {
     loadPlugins(pluginsMetadata: PluginMetadata[], container: interfaces.Container, extApi: ExtPluginApi[]): void {
         const [frontend, backend] = this.initContributions(pluginsMetadata);
         this.theiaReadyPromise.then(() => {
-            if (frontend) {
-                const worker = new PluginWorker();
-                const hostedExtManager = worker.rpc.getProxy(MAIN_RPC_CONTEXT.HOSTED_PLUGIN_MANAGER_EXT);
-                hostedExtManager.$init({
-                    plugins: pluginsMetadata,
-                    preferences: this.preferenceServiceImpl.getPreferences(),
-                    env: { queryParams: getQueryParameters() },
-                    extApi: extApi
-                });
-                setUpPluginApi(worker.rpc, container);
-                this.mainPluginApiProviders.getContributions().forEach(p => p.initialize(worker.rpc, container));
-            }
+            this.logService.provideHostLogDir().then(logHostPath => {
```

Comment:
could we replace promise/then by await code ?

Reply:
I moved this code above. If you don't like it, let me know I will use await.